### PR TITLE
[None][chore] BREAKING: Remove unused code from LlmRequest

### PR DESCRIPTION
**Pull request by Funatiq**
## Dev Engineer Review

- Removes obsolete request fields and APIs across C++, nanobind, Python, serialization, and tests.
- Deletes the unused nanobind `LlmRequest` adapter and conversion path.
- Updates constructors, serialization, bindings, and Python request creation for API consistency.
- Residual search results show related `cross_attention_mask`, `skip_cross_attn_blocks`, and lookahead identifiers outside the changed request APIs. Verify that these uses remain intentional.
- Verify external callers, serialized-request compatibility, and downstream binding imports.
- Test execution results and current review finding counts are unavailable.

## QA Engineer Review

- Updates unit tests for agent trees, KV-cache reuse, micro-batch scheduling, request serialization, request IDs, block keys, and Python bindings.
- Removes obsolete arguments and assertions. No new negative tests cover rejected legacy API usage.
- No test-list files or entries changed.
- Coverage: **needs follow-up** because execution results and integration validation are unavailable.

## Per-File QA Perspective

- `cpp/include/tensorrt_llm/batch_manager/llmRequest.h`: Removes legacy request fields and constructor parameters. Verify native callers and request construction.
- `cpp/include/tensorrt_llm/executor/executor.h`: Removes legacy `Request` APIs. Verify callers use `SamplingConfig`.
- `cpp/tensorrt_llm/executor/request.cpp`: Updates constructor forwarding and removes accessors. Verify native request creation.
- `cpp/tensorrt_llm/executor/requestImpl.h`: Removes stored legacy state. Verify validation and serialization.
- `cpp/tensorrt_llm/executor/serialization.cpp`: Deserializes fewer request fields. Verify serialized-request compatibility.
- `cpp/tensorrt_llm/nanobind/CMakeLists.txt`: Stops compiling the deleted adapter. Verify binding builds and imports.
- `cpp/tensorrt_llm/nanobind/batch_manager/bindings.cpp`: Removes obsolete Python arguments and properties. Verify supported request construction.
- `cpp/tensorrt_llm/nanobind/batch_manager/llmRequest.cpp`: Deletes tensor conversion and callback adapter code. Verify no remaining references.
- `cpp/tensorrt_llm/nanobind/batch_manager/llmRequest.h`: Deletes the nanobind wrapper. Verify downstream includes.
- `cpp/tensorrt_llm/nanobind/bindings.cpp`: Removes the deleted wrapper include. Verify binding compilation.
- `cpp/tensorrt_llm/nanobind/executor/request.cpp`: Removes legacy Python request fields. Verify request pickling and return-count behavior through `SamplingConfig`.
- `cpp/tests/unit_tests/batch_manager/agentTreeTest.cpp`: Updates agent-tree fixtures. Verify test-list coverage.
- `cpp/tests/unit_tests/batch_manager/kvCacheManagerTest.cpp`: Updates KV-cache request construction. Verify cache-reuse scenarios.
- `cpp/tests/unit_tests/batch_manager/microBatchSchedulerTest.cpp`: Removes obsolete request arguments. Verify draft-token scheduling.
- `cpp/tests/unit_tests/executor/requestTest.cpp`: Updates serialization round-trip coverage. Verify reduced request state.
- `cpp/tests/unit_tests/batch_manager/blockKeyTest.cpp`: Updates request construction. Verify block-key behavior.
- `cpp/tests/unit_tests/executor/requestWithIdTest.cpp`: Updates request construction. Verify request-ID behavior.
- `tensorrt_llm/_torch/pyexecutor/llm_request.py`: Stops passing removed arguments. Verify PyTorch executor request creation.
- `tests/unittest/bindings/test_bindings_ut.py`: Removes `pad_id` usage. Verify remaining binding properties.
- `tests/unittest/bindings/test_executor_bindings.py`: Removes obsolete request arguments. Verify multimodal and general request bindings.

## Description

BREAKING changes in internal API:

- Remove unused nanobind LlmRequest adapter.
- Remove unused fields `pad_id`, `lookahead_config`, `logits_post_processor`, `logits_post_processor_name`, `cross_attention_mask`, `num_return_sequences`, `skip_cross_attn_blocks`, `language_adapter_uid` from executor Request (C++ and Python bindings).

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: cpp/include/tensorrt_llm/executor/executor.h [hunk 1]
```diff
@@ -667,7 +666,7 @@ class Request
         std::optional<ContextPhaseParams> contextPhaseParams = std::nullopt,
         std::optional<Tensor> encoderInputFeatures = std::nullopt,
         std::optional<SizeType32> encoderOutputLength = std::nullopt,
-        std::optional<Tensor> crossAttentionMask = std::nullopt, SizeType32 numReturnSequences = 1,
+        std::optional<Tensor> crossAttentionMask = std::nullopt,
```
Comment (BowenFu) [inline cpp/include/tensorrt_llm/executor/executor.h]: `Request` is a public C++ API, so removing this positional parameter breaks existing callers; deleting the corresponding `Request::Impl` serialization slot also changes the unversioned field layout. This PR currently has neither a compatibility overload/slot nor an `api-breaking` label and `BREAKING` title. Would it be worth preserving the deprecated constructor and serialized slot for the compatibility window, or taking this through the repository's breaking-API process? I think one of those is required for this PR.

File: cpp/tests/unit_tests/batch_manager/agentTreeTest.cpp [hunk 2]
```diff
@@ -60,11 +60,10 @@ class AgentTreeTest : public ::testing::Test
         return std::make_shared<tb::LlmRequest>(requestId, maxNewTokens, inputTokens, samplingConfig, false,
             std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
             std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
-            std::nullopt, std::nullopt, std::nullopt, false, false, false, std::nullopt, std::nullopt, false,
-            std::nullopt, false, std::nullopt, false, std::nullopt, tensorrt_llm::executor::Request::kDefaultPriority,
-            std::nullopt, std::nullopt, std::nullopt, tb::LlmRequestType::LLMREQUEST_TYPE_CONTEXT_AND_GENERATION,
-            std::nullopt, 1, std::nullopt, false, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
-            agentHierarchy);
+            std::nullopt, false, false, false, std::nullopt, false, std::nullopt, false, std::nullopt,
```
[the remark this answers is not in the document]
Comment (Funatiq) [reply cpp/tests/unit_tests/batch_manager/agentTreeTest.cpp:63]: Your count is wrong. The call correctly passes 15 `std::nullopt` values. It compiles just fine.

File: cpp/include/tensorrt_llm/batch_manager/llmRequest.h [hunk 3]
```diff
@@ -111,16 +111,14 @@ class GenericLlmRequest
     using VecUniqueTokens = runtime::VecUniqueTokens;
     using BeamUniqueTokens = std::vector<VecUniqueTokens>;
     using TensorPtr = TTensor;
-    using LogitsPostProcessor = std::function<void(
-        RequestIdType, TensorPtr&, BeamTokens const&, TStream const&, std::optional<RequestIdType>)>;
     using RequestPtr = std::shared_ptr<GenericLlmRequest>;
```
Comment (zhaoyangwang-nvidia) [inline cpp/include/tensorrt_llm/batch_manager/llmRequest.h:113]: Removing `LogitsPostProcessor` leaves `TStream` (line 95) with zero uses, and deleting the nanobind wrapper removes the only instantiation that ever passed a non-default `TStream` — `GenericLlmRequest<runtime::ITensor::SharedPtr>` is now the sole instantiation in the repo. Could `TStream` be dropped here too? (`TTensor` is likewise single-valued now, but de-templatizing looks like a separate change.)

File: cpp/include/tensorrt_llm/batch_manager/llmRequest.h [hunk 4]
```diff
@@ -300,7 +273,6 @@ class GenericLlmRequest
         , mMaxNewTokens(req.getMaxTokens())
         , mSamplingConfig(req.getSamplingConfig())
         , mEndId(req.getEndId())
```
Comment (zhaoyangwang-nvidia) [inline cpp/include/tensorrt_llm/batch_manager/llmRequest.h:274]: After this hop is removed, the whole `pad_id` chain terminates in nothing: `SamplingParams.pad_id` (a documented, api-stability-protected field) → `base_worker.py:478` → `Request::Impl::mPadId` → no reader. Same for `cross_attention_mask` / `skip_cross_attn_blocks` (the Torch mllama path hardcodes `cross_attention_mask = None`) and `language_adapter_uid`. These were already inert before this PR, but it is now provable — is it worth retiring the user-facing end here, or noting it in the description as a follow-up?

File: cpp/tests/unit_tests/batch_manager/kvCacheManagerTest.cpp [hunk 5]
```diff
@@ -1229,10 +1228,9 @@ TEST_F(KVCacheManagerTest, BlockManagerReuseWithExtraIdTest)
     LlmRequest::RequestIdType requestId{0};
     auto llmRequest0 = std::make_shared<LlmRequest>(requestId, maxNewTokens, inputTokens, samplingConfig, isStreaming,
         std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
-        std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
-        std::nullopt, false, false, false, std::nullopt, std::nullopt, false, std::nullopt, false, std::nullopt, false,
-        std::nullopt, 0.5, std::nullopt, std::nullopt, std::nullopt,
-        LlmRequestType::LLMREQUEST_TYPE_CONTEXT_AND_GENERATION, inputTokenExtraIds, numReturnSequences);
+        std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, false, false,
```
Comment (zhaoyangwang-nvidia) [inline cpp/tests/unit_tests/batch_manager/kvCacheManagerTest.cpp]: These positional walls of `std::nullopt` are where removing a middle parameter can silently misalign same-typed arguments with no compiler diagnostic. I checked each site by hand and they are all correct, but since these lines are being rewritten anyway, could they adopt the `/*paramName=*/` style already used in `microBatchSchedulerTest.cpp` in this same PR?

**Review by zhaoyangwang-nvidia — APPROVED** [review]
Approve with nits.

File: cpp/include/tensorrt_llm/executor/executor.h (hunk 1 above)
Comment (zhaoyangwang-nvidia) [reply cpp/include/tensorrt_llm/executor/executor.h]: Adding a data point to this: the change also drops `nb::arg("num_return_sequences")` from the `Request` binding, so out-of-tree Python callers using `tensorrt_llm.bindings.executor` (documented in `legacy/advanced/executor.md`) get a runtime `TypeError` rather than a compile error. No in-repo caller passes it, but it is worth naming the Python-visible breakage explicitly wherever this break gets documented.

File: cpp/include/tensorrt_llm/batch_manager/llmRequest.h (hunk 3 above)
Comment (Funatiq) [reply cpp/include/tensorrt_llm/batch_manager/llmRequest.h:113]: Removed `TStream`.

File: cpp/include/tensorrt_llm/batch_manager/llmRequest.h (hunk 4 above)
Comment (Funatiq) [reply cpp/include/tensorrt_llm/batch_manager/llmRequest.h:274]: Removed a bunch of unused fields from the executor Request.

File: cpp/tests/unit_tests/batch_manager/kvCacheManagerTest.cpp (hunk 5 above)
Comment (Funatiq) [reply cpp/tests/unit_tests/batch_manager/kvCacheManagerTest.cpp]: Added parameter-name comments.

File: cpp/include/tensorrt_llm/executor/executor.h (hunk 1 above)
Comment (Funatiq) [reply cpp/include/tensorrt_llm/executor/executor.h]: Added BREAKING changes in the PR description.

File: cpp/tensorrt_llm/nanobind/executor/request.cpp [hunk 6]
```diff
@@ -568,19 +565,17 @@ void initRequestBindings(nb::module_& m)
...
-            self.getEncoderInputFeatures(), self.getEncoderOutputLength(), self.getCrossAttentionMask(),
-            self.getSkipCrossAttnBlocks(), self.getGuidedDecodingParams(), self.getDisaggRequestId(),
-            self.getCacheSalt());
+            self.getEncoderInputFeatures(), self.getEncoderOutputLength(), self.getGuidedDecodingParams(),
+            self.getDisaggRequestId(), self.getCacheSalt());
     };
     auto requestSetstate = [](tle::Request& self, nb::tuple const& state)
     {
-        if (state.size() != 33)
+        if (state.size() != 27)
```
[the remark this answers is not in the document]
Comment (Funatiq) [reply cpp/tensorrt_llm/nanobind/executor/request.cpp:578]: Mixed-version does not need to be supported.

**Review by eopXD — APPROVED** [review]

**Review by BowenFu — APPROVED** [review]

**Review by arysef — APPROVED** [review]
